### DISCUSSION TITLE
Detection of the number of speakers in Speaker diarization pipeline

### DISCUSSION THREAD

**Discussion by Vanargh** [pyannote/pyannote-audio#779] [Q&A] [answered]
I am also facing another major issue.

With the pipeline

`pipeline = torch.hub.load('pyannote/pyannote-audio', 'dia')` 

the segmentation and detection of speech segments are extremely good with very fine-grained segments. But with regards to the number of speakers in audio, the detection of the speakers is limited to 2-3 at max.. For audio files with a large number of speakers (say more than 5), the results obtained after applying the pre-trained pipeline still show 2 or 3 speakers. Have I incorrectly applied the pre-trained pipeline to obtain such results?
If not, is there some way this problem can be solved to obtain better results?

**Comment by hbredin** [comment, marked as answer]
You have to tune the hyper parameters of the pipeline on your own validation data. See related tutorial.

**Reply by Vanargh** [reply]
Thank you @hbredin. In the tutorials presented in `pyannote.audio.tutorials`,  it is either fine-tuning or training from scratch a single module (eg. SAD). How do you fine-tune hyper-parameters of the entire pre-trained pipeline of speaker diarization? Is there some link or tutorial provided for the same? Or do we have to individually fine-tune hyper-parameters for each component (SAD, SCD etc.) on our local system and combine them?

**Reply by hbredin** [reply]
https://github.com/pyannote/pyannote-audio/tree/master/tutorials/pipelines/speaker_diarization

**Reply by dwarkeshsp** [reply]
This link no longer works and I can't find a notebook with a similar title in Tutorials. Where should I be looking for a useful resource on how one might figure out number of speakers (aka number of clusters of embeddings)?
